test/seqan3-test.cmake

```diff
@@ -26,6 +26,20 @@ option (SEQAN3_TEST_BUILD_OFFLINE "Skip the update step of external projects." O
 # have to be adapted or the option deactivated.
 option (SEQAN3_BENCHMARK_ALIGN_LOOPS "Pass -falign-loops=32 to the benchmark builds." ON)
 
+# ----------------------------------------------------------------------------
+# Custom Build types
+# ----------------------------------------------------------------------------
+
+# -DCMAKE_BUILD_TYPE=FEDORA; our library did not compile for fedora quite a few times, because of that we created this
+# custom build type to emulate their flag set-up.
+# We omitted:
+#   -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1
+#   -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1
+#   set -mtune=native
+#   and -fcf-protection=check
+# See https://src.fedoraproject.org/rpms/redhat-rpm-config/blob/rawhide/f/buildflags.md for an overview
+set (CMAKE_CXX_FLAGS_FEDORA "-O2 -flto -ffat-lto-objects -fexceptions -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -fstack-protector-strong -m64 -mtune=native -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection=check")
```

Comment:
Also, to what do the other flags default, nothing? They should probably be debug?
You could "inherit" fromd debug, e.g. `set (CMAKE_C_FLAGS_FEDORA "${CMAKE_C_FLAGS_DEBUG")`, also for the linker flags.

On the nightlies, I use Debug and just add the cxx flags

Reply:
Sorry, I don't quite understand that comment. I think we don't need the C flags, because this should be a C++ project.

> On the nightlies, I use Debug and just add the cxx flags

And we could change that to `-DCMAKE_BUILD_TYPE=FEDORA`

Reply:
It might be a difference if you build in debug and add some cxxflags; or change the build type. E.g. C flags would be the debug flags, etc.; linker flags would also be debug flags.
Now there are probably no flags for linker or C (gtest uses C, no?); because we do not set them.

Reply:
ah; I thought of fedora like a specialised RELEASE build type

Reply:
That's ok, but I'm not sure what with the other flags happens. 
We can either leave it as is, or define the C and linker flags, even if we just alias the release flags.

Reply:
the other flags are empty. you don't need to specialise them.